libs/coin-modules/coin-tezos/src/logic/listOperations.ts

```diff
@@ -24,17 +30,21 @@ export async function listOperations(
   { lastId, limit }: { lastId?: number; limit?: number },
 ): Promise<[Operation[], number]> {
   const operations = await tzkt.getAccountOperations(address, { lastId, limit });
-  return [
+  const operationswithBlock = await Promise.all(
     operations
       .filter(op => isAPITransactionType(op) || isAPIDelegationType(op))
-      .reduce((acc, op) => acc.concat(convertOperation(address, op)), [] as Operation[]),
-    operations.slice(-1)[0].id,
-  ];
+      .map(async op => {
+        const block = await tzkt.getBlockByHash(op.block);
```

Comment:
This will trigger hundreds of request each time we want the `listOperations`.
There is no other way to retrieve the info?

Reply:
As per the suggestion below, I'd also suggest removing this, unless we need other block-specific information?

For reference here's what the block info call returns

```
{
    "cycle": 802,
    "level": 7176273,
    "hash": "BMKnKkbSALea5sDqb3YdyRwGiEKT22Qm1Gw5ZKvKXBZLTpaEbjw",
    "timestamp": "2024-11-21T17:04:20Z",
    "proto": 20,
    "payloadRound": 0,
    "blockRound": 0,
    "validations": 6989,
    "deposit": 0,
    "rewardDelegated": 3038694,
    "rewardStakedOwn": 1402472,
    "rewardStakedEdge": 1055150,
    "rewardStakedShared": 0,
    "bonusDelegated": 3021927,
    "bonusStakedOwn": 1394734,
    "bonusStakedEdge": 1049327,
    "bonusStakedShared": 0,
    "fees": 179996,
    "nonceRevealed": false,
    "proposer": {
        "address": "tz3UC5hM9uCUiqmAw1pHdxjAdGueQMd8u3wf"
    },
    "producer": {
        "address": "tz3UC5hM9uCUiqmAw1pHdxjAdGueQMd8u3wf"
    },
    "software": {
        "date": "2024-11-14T23:12:25Z"
    },
    "lbToggleEma": 446596880,
    "aiToggleEma": 181960867,
    "rewardLiquid": 3038694,
    "bonusLiquid": 3021927,
    "reward": 5496316,
    "bonus": 5465988,
    "priority": 0,
    "baker": {
        "address": "tz3UC5hM9uCUiqmAw1pHdxjAdGueQMd8u3wf"
    },
    "lbEscapeVote": false,
    "lbEscapeEma": 446596880
}
```

Reply:
For the moment we need only the time, the hash and the height of the block, nothing else. So I removed this call like you suggested